xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.h

```diff
@@ -78,6 +81,18 @@ struct VideoPicture
   unsigned int iDisplayWidth;           //< width of the picture without black bars
   unsigned int iDisplayHeight;          //< height of the picture without black bars
 
+  bool operator==(const VideoPicture& pic) const
+  {
+    return this->iWidth == pic.iWidth &&
+           this->iHeight == pic.iHeight &&
+           this->iDisplayWidth == pic.iDisplayWidth &&
+           this->iDisplayHeight == pic.iDisplayHeight &&
+           this->stereoMode == pic.stereoMode &&
+           this->color_primaries == pic.color_primaries &&
+           this->color_transfer == pic.color_transfer &&
+           this->hdrType == pic.hdrType;
+  }
```

Comment:
In C++20 exist default operator:

`bool operator==(const VideoPicture& pic) const = default;`

But we don't want this because we don't want to compare all members (pts, dts and others change). Then is implemented custom operator ==. Nothing seems wrong to me.

>  it will be easier to forget to add a new field when needed

I don't think that's a compelling reason --> Forgot for years anyway with old code

> The !m_pRenderer->ConfigChanged(picture) could be included as well (is CWinRenderer::ConfigChanged used from anywhere else? could be removed)
> 
> VideoPicture has more fields than necessary and troublesome semantics due to the videoBuffer pointer, an alternative could be a new renderer struct type, but I don't care much.

Seems things for other PR.

Reply:
> it will be easier to forget to add a new field when needed

New fields are added to `VideoPicture` struct so it is better to do the comparison there precisely because then the changes are all done in the same place (if something is added it should be checked if it needs to be initialized, reset, compared, etc.)

Reply:
I know we don't want the default operator for this PR and it may not be desirable even outside of that scope (behavior with pointers).

Speaking of, is there any existing code using the VideoPicture == operator? It would start behaving differently.

I'm not in the habit of questioning whether == tests some or all the members when reading or writing code, but if other team members are fine with it, so am I.

Reply:
>> it will be easier to forget to add a new field when needed

> New fields are added to VideoPicture struct so it is better to do the comparison there precisely because then the changes are all done in the same place (if something is added it should be checked if it needs to be initialized, reset, compared, etc.)

Agreed, but see comment about existing code. Changing the operator has global impact. I'm not aware of tools that help you locate the operator users. Text search won't do.

Reply:
> Speaking of, is there any existing code using the VideoPicture == operator? It would start behaving differently.

bool operator (in VideoPicture) it has implemented in this PR (not existing before)

How could it be possible that it was being used before?

Reply:
Yes you're right, new with C++20 and needs explicit definition even with =default.